client/client.js

```diff
@@ -0,0 +1,162 @@
+var log = require('minilog')('radar:client'),
+    Core = require('../core');
+
+function Client (name, id) {
+  this.createdAt = Date.now();
+  this.timestamp = Date.now();
+  this.name = name;
+  this.id = id;
+  this.subscriptions = {};
+  this.presences = {};
+};
+
+// 86400000:  number of milliseconds in 1 day
+var DEFAULT_DATA_TTL = 86400000;
+
+// 900000:  number of milliseconds in 15 minutes
+var DEFAULT_DATA_PERIOD = 900000;
+
+// Class properties
+Client.serverName;
+Client.clients = {};
+Client.names = {}
+
+require('util').inherits(Client, require('events').EventEmitter);
+
+// Public API
+
+// Class methods
+Client.serverNameSet = function (serverName) {
+  Client.serverName = serverName;
+};
+
+Client.nameSet = function(id, name) {
+  Client.names[id] = name;
+};
+
+// For a given client id/name pair, store the associated message
+Client.dataStore = function (id, message) {
+  var name;
+  if (Client.names[id]) {
+    name = Client.names[id];
+  }
+  else {
+    log.error('client id:', id, 'does not have corresponding name');
+    return false;
+  }
+
+  // Fetch the current instance, or create a new one
+  var client = Client._get(id, name);
+
+  subscriptions = client.subscriptions;
+  presences = client.presences;
+
+  // Persist the message data, according to type
+  var changed = true;
+  switch(message.op) {
+    case 'unsubscribe':
+      if (subscriptions[message.to]) {
+        delete subscriptions[message.to];
+      }
+      break;
+
+    case 'sync':
+    case 'subscribe':
+      client.timestamp = Date.now();
+      subscriptions[message.to] = message.op;
+      break;
+
+    case 'set':
+      client.timestamp = Date.now();
+      if (message.to.substr(0, 'presence:/'.length) == 'presence:/') {
+        presences[message.to] = message.value;
+      }
+      break;
+
+    default:
+      changed = false;
+  }
+
+  // TODO: perhaps do this every N "stores"
+  if (changed) {
+    Core.Persistence.persistHash(Client.serverName, name, client);
```

Comment:
I think the key should be radar_client:/<name> instead. Because servers may die and we may need to look up the client from another server.